pkg/storage/azureblob/azureblob.go

```diff
@@ -0,0 +1,155 @@
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	pipe := azblob.NewPipeline(cred, azblob.PipelineOptions{})
+	serviceURL := azblob.NewServiceURL(*accountURL, pipe)
+	// rules on container names:
+	// https://docs.microsoft.com/en-us/rest/api/storageservices/naming-and-referencing-containers--blobs--and-metadata#container-names
+	//
+	// This container must exist
+	containerURL := serviceURL.NewContainerURL(containerName)
+	cl := &azureBlobStoreClient{containerURL: &containerURL, timeout: timeout}
+	return cl, nil
+}
+
+// Storage implements (github.com/gomods/athens/pkg/storage).Saver and
+// also provides a function to fetch the location of a module
+type Storage struct {
+	client client
+}
+
+// New creates a new azure blobs storage
+func New(conf *config.AzureBlobConfig, timeout time.Duration) (*Storage, error) {
+	const op errors.Op = "azure.New"
+	u, err := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net", conf.AccountName))
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	cl, err := newBlobStoreClient(u, conf.AccountName, conf.AccountKey, conf.ContainerName, timeout)
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	return &Storage{client: cl}, nil
+}
+
+// BlobExists checks if a particular blob exists in the container
+func (c *azureBlobStoreClient) BlobExists(ctx context.Context, path string) (bool, error) {
+	const op errors.Op = "azure.BlobExists"
+	// TODO: Any better way of doing this ?
+	blobURL := c.containerURL.NewBlockBlobURL(path)
+	_, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{})
+	if err != nil {
+		var serr azblob.StorageError
+		var ok bool
+
+		if serr, ok = err.(azblob.StorageError); !ok {
+			return false, errors.E(op, fmt.Errorf("Error in casting to azure error type %v", err))
+		}
+		if serr.Response().StatusCode == http.StatusNotFound {
+			return false, nil
+		}
+
+		return false, errors.E(op, err)
+	}
+	return true, nil
+
+}
+
+// ReadBlob returns an io.ReadCloser for the contents of a blob
+func (c *azureBlobStoreClient) ReadBlob(ctx context.Context, path string) (io.ReadCloser, error) {
+	const op errors.Op = "azure.ReadBlob"
+	blobURL := c.containerURL.NewBlockBlobURL(path)
+	downloadResponse, err := blobURL.Download(ctx, 0, 0, azblob.BlobAccessConditions{}, false)
+	if err != nil {
+		return nil, errors.E(op, err)
+	}
+	return downloadResponse.Body(azblob.RetryReaderOptions{}), nil
+}
+
+func (c *azureBlobStoreClient) ListBlobs(ctx context.Context, prefix string) ([]string, error) {
+	const op errors.Op = "azure.ListBlobs"
+	var blobs []string
+	for marker := (azblob.Marker{}); marker.NotDone(); {
+		listBlob, err := c.containerURL.ListBlobsFlatSegment(ctx, marker, azblob.ListBlobsSegmentOptions{
+			Prefix: prefix,
+		})
+		if err != nil {
+			return nil, errors.E(op, err)
+		}
+		marker = listBlob.NextMarker
+
+		for _, blob := range listBlob.Segment.BlobItems {
+			blobs = append(blobs, blob.Name)
+		}
+	}
+	return blobs, nil
+}
```

Comment:
Could you add new line here?

Reply:
Sure, but I was wondering why we are doing it, can you help me out?

Reply:
Just code formating, nothing major

Reply:
Oh okay, cool 😃 Fixing it now 🔧